utils/model_uploader/model_autotracing.py

```diff
@@ -0,0 +1,452 @@
+        ), f"Raised Exception in tracing {model_format} model\
+                             during initiating a sentence transformer model class object: {e}"
+
+    # 2.) Save the model in the specified format
+    model_path = None
+    try:
+        if model_format == TORCH_SCRIPT_FORMAT:
+            model_path = pre_trained_model.save_as_pt(
+                model_id=model_id, sentences=TEST_SENTENCES
+            )
+        else:
+            model_path = pre_trained_model.save_as_onnx(model_id=model_id)
+    except Exception as e:
+        assert False, f"Raised Exception during saving model as {model_format}: {e}"
+
+    # 3.) Create a model config json file
+    try:
+        pre_trained_model.make_model_config_json(
+            version_number=model_version,
+            model_format=model_format,
+            embedding_dimension=embedding_dimension,
+            pooling_mode=pooling_mode,
+        )
+    except Exception as e:
+        assert (
+            False
+        ), f"Raised Exception during making model config file for {model_format} model: {e}"
+
+    # 4.) Return model_path & model_config_path for model registration
+    model_config_path = folder_path + MODEL_CONFIG_FILE_NAME
+    return model_path, model_config_path
+
+
+def register_and_deploy_sentence_transformer_model(
+    ml_client: "MLCommonClient",
+    model_path: str,
+    model_config_path: str,
+    model_format: str,
+) -> List["DTypeLike"]:
+    """
+    Register the pretrained sentence transformer model by using the model file and the model config file,
+    deploy the model to generate embeddings for the TEST_SENTENCES,
+    and return the embeddings for model verification
+
+    :param ml_client: A client that communicates to the ml-common plugin for OpenSearch
+    :type ml_client: MLCommonClient
+    :param model_path: Path to model file
+    :type model_path: string
+    :param model_config_path: Path to model config file
+    :type model_config_path: string
+    :param model_format: Model format ("TORCH_SCRIPT" or "ONNX")
+    :type model_format: string
+    """
+    embedding_data = None
+
+    # 1.) Register & Deploy the model
+    model_id = ""
+    try:
+        model_id = ml_client.register_model(
+            model_path=model_path,
+            model_config_path=model_config_path,
+            deploy_model=True,
+            isVerbose=True,
+        )
+        print()
+        print(f"{model_format}_model_id:", model_id)
+        assert model_id != "" or model_id is not None
+    except Exception as e:
+        assert (
+            False
+        ), f"Raised Exception in {model_format} model registration/deployment: {e}"
+
+    # 2.) Check model status
+    try:
+        ml_model_status = ml_client.get_model_info(model_id)
+        print()
+        print("Model Status:")
+        print(ml_model_status)
+        assert ml_model_status.get("model_format") == model_format
+        assert ml_model_status.get("algorithm") == "TEXT_EMBEDDING"
```

Comment:
We need to check if model is DEPLOYED or not

Reply:
<img width="588" alt="Screenshot 2023-07-24 at 11 32 52 AM" src="https://github.com/opensearch-project/opensearch-py-ml/assets/106889996/490aa03c-91d1-4840-974d-16f431969782">

Doesn't this line already handle that?

Reply:
Technically I agree. But in the `Check model status` status asserting model status kinda makes more sense to me. Also what'll happen to the workflow if any of this code throws the exception?

Reply:
Sure! I can add another check. If any of the code throws the exception, the workflow will exit with failed status.